src/providers/RedisProvider.js

```diff
@@ -0,0 +1,201 @@
+const Provider = require('./Provider');
+const promisify = require("util").promisify;
```

Comment:
For the sake of consistency
```suggestion
const { promisify } = require("util");
```